data/transactions/logic/eval_test.go

```diff
@@ -1806,6 +1806,196 @@ func testLogic(t *testing.T, program string, v uint64, ep EvalParams, problems .
 	}
 }
 
+func TestGtxnaScratch(t *testing.T) {
+	t.Parallel()
+
+	type scratchTestCase struct {
+		tealSources []string
+		errContains string
+	}
+
+	simpleCase := scratchTestCase{
+		tealSources: []string{
+			`int 2
+store 0
+int 1`,
+			`gtxna 0 Scratch 0
```

Comment:
Assuming we commit to allow only static offsets in Scratch.

Reply:
Why only static?

Reply:
I meant "constant" - because group index and scratch position are constant in `gtxna 0 Scratch 0` opcode and need to known on assembly time. This might by OK, not sure about real use cases (like passing gi and pos as app call arguments).

Reply:
Oh I see. The group index can be dynamic since I only modified the `txnFieldToStack` function, which I believe is used by the `gtxnsa` logic. But that's a good point that the Scratch index will have to be static.

Reply:
Yes, that will make it available to all of txna, gtxna, gtxnsa.

I think the scratch slot is okay static for now, since we have the same limitation load load/store.